satchless/cart/tests/__init__.py

```diff
@@ -142,96 +126,18 @@ def test_basic_cart_ops(self):
-        remove_item_url = cart_app.reverse('remove-item', args=(cart.items.get().id,))
-        response = self._test_status(remove_item_url, method='post',
-                                     status_code=302, client_instance=self.client)
-        self.assertRedirects(response, cart_app.reverse('details'))
+        cart.add_item(self.macaw_blue, -101)
+        self.assertEqual(cart.get_quantity(self.macaw_blue), Decimal('0'))
+        self.assertEqual(cart.total(), Price(2520, currency='USD'))
 
-    def test_cart_view_with_item(self):
-        cart = self._get_or_create_cart_for_client(self.client)
-        cart.replace_item(self.macaw_blue_fake, Decimal('2.45'))
-        self._test_status(cart_app.reverse('details'),
-                          client_instance=self.client, status_code=200)
-
-    def test_cart_view_updates_item_quantity(self):
-        cart = self._get_or_create_cart_for_client(self.client)
-        cart.replace_item(self.macaw_blue_fake, Decimal(1))
-        response = self._test_status(cart_app.reverse('details'),
-                                    client_instance=self.client, status_code=200)
-        cart_item_form = response.context['cart_item_forms'][0]
-        data = {
-            'quantity': 2
-        }
-        data = dict((cart_item_form.add_prefix(key), value) for (key, value) in data.items())
-        self._test_status(cart_app.reverse('details'), data=data,
-                          method='post', status_code=302,
-                          client_instance=self.client)
-        self.assertEqual(cart.items.count(), 1)
-        self.assertEqual(cart.items.all()[0].quantity, 2)
-
-    def test_add_by_view_for_anonymous(self):
-        cli_anon = Client()
-        self._test_add_by_view(cli_anon)
-
-    def test_add_by_view(self):
-        cli_user1 = Client()
-        self.assertTrue(cli_user1.login(username="testuser", password=u"pasło"))
-        self._test_add_by_view(cli_user1)
-
-    def test_add_to_cart_form_handles_incorrect_data(self):
-        cli_anon = Client()
-        response = self._test_status(self.macaw.get_absolute_url(),
-                                     method='post',
-                                     data={'typ': 'cart',
-                                           'color': 'blue',
-                                           'looks_alive': 1,
-                                           'quantity': 'alkjl'},
-                                     client_instance=cli_anon,
-                                     status_code=200)
-        errors = response.context['product'].cart_form.errors
-        self.assertTrue('quantity' in errors)
+    def _get_or_create_cart_for_client(self, client=None, typ='cart'):
+        try:
+            return TestCart.objects.get(
+                pk=client.session[CART_SESSION_KEY % typ])[0]
+        except KeyError:
+            cart = TestCart.objects.create(typ=typ)
+            client.session[CART_SESSION_KEY % typ] = cart.pk
+            return cart
```

Comment:
This doesn't seem to be used. It's also defined in `checkout` which does have one caller though.